DP Mystery Map/Assets/Scripts/PlayerController.cs

```diff
@@ -208,13 +222,83 @@ private void GridMoveUpdate()
                 Player.FacingDirection = _walkDirection;
             }
 
+            //Update the idleFacing position per direction
+            switch (Player.FacingDirection)
+            {
+                case Direction.Up:
+                    playerAnimator.SetFloat("faceIdleX", 0);
+                    playerAnimator.SetFloat("faceIdleY", -1);
+                    break;
+                case Direction.Down:
+                    playerAnimator.SetFloat("faceIdleX", 0);
+                    playerAnimator.SetFloat("faceIdleY", 1);
+                    break;
+                case Direction.Left:
+                    playerAnimator.SetFloat("faceIdleX", -1);
+                    playerAnimator.SetFloat("faceIdleY", 0);
+                    break;
+                case Direction.Right:
+                    playerAnimator.SetFloat("faceIdleX", 1);
+                    playerAnimator.SetFloat("faceIdleY", 0);
+                    break;
+            }
+
             CheckKeys();
         }
         //else check for key being pressed down
         else
         {
             CheckKeys();
         }
+
+        //Handle direction faced while moving and idle
+            if (_isPlayerMoving == false)
+            {
+                playerAnimator.SetFloat("Horizontal", playerMovementController.x);
+                playerAnimator.SetFloat("Vertical", playerMovementController.y);
+                playerAnimator.SetFloat("Speed", playerMovementController.sqrMagnitude);
+            }
+            else
+            {
+            //Enable running animation if shift is held
+            if (Input.GetKey(KeyCode.LeftShift) && ((Input.GetKey(Player.MoveUpKey)) ||
+                (Input.GetKey(Player.MoveDownKey)) || (Input.GetKey(Player.MoveLeftKey)) ||
+                (Input.GetKey(Player.MoveRightKey))))
+            {
+                playerAnimator.SetBool("isRunning", true);
+                GridWalkDuration = 0.15625f;
+            }
+            else
+            {
+                playerAnimator.SetBool("isRunning", false);
+                GridWalkDuration = 0.25f;
+            }
+
+            switch (_walkDirection)
+                {
+                    case Direction.Left:
+                        playerMovementController.x = Input.GetAxisRaw("Horizontal");
```

Comment:
Don't use `GetAxisRaw()`. If the player presses the left arrow or A, then `GetAxisRaw("Horizontal")` returns -1. Right arrow or D returns 1. And so on. `GetAxisRaw` won't work properly if the player changes the movement keys. Just set the Vector2 to the predefined static Vector2 (e..g, `Vector2.left`).
Put this in a new `DirectionChangeEventHandler` and subscribe it to the `Player.directionChangeEvent` event. Make sure to unsubscribe the event in the `PlayerController.OnDestroy()` method too. **You may consider putting all of the animator event subscribers in one function.**